source/beamline/BioXAS/BioXAS32ElementGeDetector.cpp

```diff
@@ -31,3 +34,42 @@ bool BioXAS32ElementGeDetector::setAcquisitionTime(double seconds)
 
 	return true;
 }
+
+AMAction3* BioXAS32ElementGeDetector::createEraseAction()
+{
+	AMAction3 *result = 0;
+
+	// Build the erase action.
+
+	AMListAction3* eraseAction = new AMListAction3(new AMListActionInfo3("Erase detector data", "Erase detector data"), AMListAction3::Sequential);
+
+	// Begin by manually toggling the erase control.
+
+	eraseAction->addSubAction(AMActionSupport::buildControlMoveAction(eraseControl_, 1.0));
+	eraseAction->addSubAction(AMActionSupport::buildControlMoveAction(eraseControl_, 0.0));
```

Comment:
Ln44 ~ Ln49 can be :
`AMListAction3* eraseAction = AMXspress3XRFDetector::createEraseAction();`